### ISSUE TITLE
Allow inline return types between parameters and code block

### ISSUE THREAD

**Issue by shicks** [google/closure-compiler#1306]
```
class Foo {
  bar(/** string */ arg) /** number */ { return arg.length + 1; }
  get baz() /** string */ { return 'xyz'; }
}
```

This would be a reasonable way to write return types, and mirrors typescript syntax pretty well (as well as other languages, like Go), so it shouldn't be too surprising.

**Comment by dimvar** [comment]
Inline types are not a super high priority ATM. Was there something specific that prompted this?

**Comment by blickly** [comment]
Note that historically we have put the inline return type before the function name:
http://closure-compiler-debugger.appspot.com/#input0%3Dfunction%2520%252F**%2520number%2520*%252F%2520f()%2520%257B%2520return%25205%253B%2520%257D%253B%250Avar%2520%252F**%2520string%2520*%252F%2520s%2520%253D%2520f()%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by trevorade** [comment]
I apologize for the the [thread necromancy](https://tvtropes.org/pmwiki/pmwiki.php/Main/ThreadNecromancer) but I was curious if there has been any additional development on this in the past few years?

Inlining parameter types is becoming more common but I feel it's blocked on this issue.

Probably a separate issue but is inlining the return type possible for arrow functions?

**Comment by brad4d** [comment]
I don't believe there's been any work toward adding this new form of inline annotation.

It should work to put an inline annotation just before an arrow function,
but it currently doesn't.

https://closure-compiler-debugger.appspot.com/#input0%3Dlet%2520f%2520%253D%2520%252F**%2520number%2520*%252F%2520()%2520%253D%253E%25205%253B%250Avar%2520%252F**%2520string%2520*%252F%2520s%2520%253D%2520f()%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3Dtrue%26CHECK_TYPES%3Dtrue%26CHECK_SYMBOLS%3Dtrue%26MISSING_PROPERTIES%3Dtrue%26TRANSPILE%3Dtrue%26PRESERVE_TYPE_ANNOTATIONS%3Dtrue%26PRETTY_PRINT%3Dtrue
